### ISSUE TITLE
Animation Tool: Visualize simulation results on the map canvas

### ISSUE THREAD

**Issue by leendertvanwolfswinkel** [nens/threedi-results-analysis#666]
This ticket re-implements the functionality of the Animation Toolbar. The new implementation is so much more integrated with other parts of the UI, that the animation toolbar _as a separate tool_ will disappear:

The current implementation of the Animation Toolbar has the following UI components:
- Time slider. Instead, we will now use the native QGIS Temporal Controller
- Layers. Instead, we will now use the layers of the computational grid, dynamically changing their styling.
- Toolbar/user interface. A toolbar is too small for the UI that is needed for the new "multiple results" functionality (switching the visualisation for multiple grids and choosing between different result sets). The results manager window has more space for such options.

Tasks for this ticket:
- [x] In the results manager, users can choose which simulation result to visualize for each computational grid, and which variable to use. They can also choose not to visualize any result, so they can inspect the details of the computational grid itself
- [x] Users can set the current time in the native QGIS Temporal Controller, and the result for that timestep will be visualized on the map canvas
- [x] If a user identifies on a flowline/node/pump line/cell, they should see the value (discharge/water level/etc, depending on which variable has been selected in the results manager) for each simulation result. So for example, I have loaded results for T10, T25 and T100, set the flowline visualisation to 'Discharge' and the temporal controller to 2017-08-23 12:00. I use the Identify tool to click on a flowline and I get an overview of its (static) attributes, plus three attributes 't10', 't25', 't100', each showing the discharge through that flowline on 2017-08-23 12:00 for the different rain events. Suggested implementation is to add _virtual fields_ to the layers of the computational grid, one for each simulation result, and update its values when the variable is changed / when the time slider moves.
- [x] Should also be able to show results from the aggregation netcdf
- [x] When the first simulation result is added to a computational grid, the default visualisation for that grid should be "Net cumulative discharge" (total discharge for each flowline, read from the aggregate results netcdf) and the time should be set to the end of the simulation. The styling (bins) should be based on the value distribution in the result set, as in the current implementation of the Animation Toolbar. 
- [x] Users can add multiple result sets to each computational grid

**Comment by benvanbasten-ns** [comment]
- [x] Refactor all animation specific code from main module into anim module
- [x] First restore old functionality with new datamodel / UI
- [x] Move anim UI to results manager UI
- [x] Extend model with checked results and properties for number of grids and results
- [x] Only one result per grid can be selected
- [x] Result selection needs to be saved
- [x] model serialization to separate module 
- [x] When grid is removed, also emit result_removed when applicable
- [x] Dynamically adding or removing fields when result is added/removed
- [x] Dynamically set value of fields when different timekey or param is selected
- [x] Dynamically change styling (look at different field) when result is selected/added/removed or parameter changed (bounds). Revert to default style when all results are deselected.
- [x] Relative animation checkbox needs to be verified
- [x] Additional fields need to be added to layers (e.g. z_coordinate)
- [x] ~Check velocity_thresholds styling~
- [x] Check renaming of grid layer group
- [x] Fix retrieval of aggregate variables
- [x] Change default parameter setting when result is selected for first time?
- [x] Disable anim UI when no results selected

**Update for multiple grids and results:**

- [x] Animate multiple results (for-loop in MapAnimator)
- [x] Parameters in combobox (union of parameters in result files? or intersection?)
- [ ] Temporal extent of temporal controller should be set to union of durations of selected results

**Comment by benvanbasten-ns** [comment]
@leendertvanwolfswinkel In the code sometimes a distinction is made between results containing "groundwater" and those that do not, is that still required or can be assume all results contain groundwater?

**Comment by arjanverkerk** [comment]
Can someone point to the _virtual fields_ in any QGIS docs? We found _expression fields_, but they are derived properties based on other attributes, as far as I can tell. We can set this expression dynamically, but I have not been able to figure out how this can be used to bicycle arbitrary python arrays into the field. There are _virtual layers_ but that is a different concept which is a layer whose features are actually the result of an SQL query on (vector)datasources of other layers.

**Comment by benvanbasten-ns** [comment]
@leendertvanwolfswinkel @arjanverkerk They are the same. Virtual fields are also named Expression Fields in QGIS.

It is possible to have a custom piece of code executed (https://docs.qgis.org/3.22/en/docs/user_manual/expressions/expression.html#function-editor) but I'm not sure whether this is possible via the QGIS API. 

As we only want to update the virtual fields when the temporal controller changes (for which we have a signal), I suggest to just loop over the features and set the values in the virtual fields. I think that the previous implementation also looped over the features.

**Comment by leendertvanwolfswinkel** [comment]
Most results do not contain groundwater results. Please keep that distinction.

**Comment by arjanverkerk** [comment]
I found we can only assign only one expression to the whole column. That expression will be evaluated whenever the attribute value is retrieved. But I also found we can define our own python functions for use with the expression, and therefore use any python function in the expression. All nice, but the performance is not good enough. An expression which involves looking up a qgs-registered function is much slower then an expression that just contains a number "1.2".
